org-rainbow-tags.el

```diff
@@ -245,39 +268,74 @@ background by adding `:inverse-video t' to
     (eval (org-rainbow-tags--set-face face-name color))
     face-name))
 
-(defvar org-rainbow-tags-filetags-regexp
-  (rx (seq
-       line-start
-       "#+filetags:"
-       (one-or-more whitespace)
-       ;; Capture the rest of the line
-       (group-n 1 (zero-or-more any))))
-  "Regular expression to match Org-mode filetags lines.")
-
-(defun org-rainbow-tags--apply-overlay-to-match ()
-  "Apply the auto-generated tag faces to the current regex match."
-  (goto-char (match-beginning 0))
-  (while (re-search-forward org-tag-group-re (line-end-position) t)
-    (when (eolp)
-      (save-excursion
-        (goto-char (match-beginning 0))
-        (while (re-search-forward org-rainbow-tags--org-tag-regexp (line-end-position) t)
-          (let* ((overlay (make-overlay (match-beginning 1) (match-end 1))))
-            (overlay-put overlay 'face (org-rainbow-tags--get-face 1))
-            (add-to-list 'org-rainbow-tags--overlays overlay)
-            (backward-char 2)))))))
+(defun org-rainbow-tags--apply-overlay (beg end)
+  "Create and apply `org-rainbow-tags' overlay to the text between BEG and END.
+
+All created overlays are added to the `org-rainbow-tags--overlays' list for tracking purposes."
+  (let ((overlay (make-overlay beg end)))
+    (overlay-put overlay 'face (org-rainbow-tags--get-face 1))
+    (push overlay org-rainbow-tags--overlays)))
+
+(defun org-rainbow-tags--apply-overlays-to-matched-tag-section ()
+  "Apply a font face overlay to matched tag sections in the current buffer.
+
+For dynamic blocks containing clocktables, the function will apply overlays
+to each tag in each cell of the clocktable column with the header `Tags'.
+
+For normal Org tag groups (e.g., ':tag1:tag2:'), the function will apply overlays
+to each tag in the group."
+  (cond
+   ;; Case for dynamic blocks that contain clocktables
+   ((looking-at-p org-dblock-start-re)
+    (let ((table-block-end (save-excursion
+                             (when (re-search-forward org-dblock-end-re (point-max) t)
+                               (beginning-of-line)
+                               (point))))
+          (tags-column-start nil)
+          (end-of-field nil))
+      (when (and table-block-end
+                 (re-search-forward org-table-hline-regexp table-block-end t))
+        (forward-line -1)
+        (when (search-forward "Tags" (line-end-position) t)
+          (search-backward "|")
+          (setq tags-column-start (current-column))
+          ;; Using these two is so much faster than `next-line' that I feel
+          ;; like I have to be missing a built-in that does this.
+          (forward-line 1)
+          (move-to-column tags-column-start)
```

Comment:
`(line-move 1)` maybe? I didn't check the performance though.

Reply:
I don't have time to test right now, but I would bet not. `forward-line` and `move-to-column` are both C functions, and `line-move` is another Elisp one like `next-line`. `line-move` is used by `next-line`, so it might be a bit faster to use it directly, but the code for `line-move` is looks a lot more involved than `next-line`.